### PR TITLE
Migrate Edraak certificates 

### PR DESCRIPTION
**Pull request by OmarIthawi**
### Description

TASK: [Migrate Edraak's PDF certificate with tests](https://app.asana.com/0/256944342521670/395647354508655)

### TODO
 - [x] Migrate the views with a feature flag
 - [x] Migrate styles
 - [x] Allow using logos from the database (document how to do it)
 - [x] Refactor (and test) the `return redirect(reverse('dashboard'))` in the cert view to 404 error!
 - [x] ~~Take a look at **edX Persistent Grades**. It could be an alternative `cached_is_course_passed` function.~~ not a problem! We can definitely live with it!

#### Unit tests
 - [x] Tests for the helper functions and other stuff
 - [x] Prevent adding new files to the `assets`

#### Integration tests
 - [x] Staff (where the checks are usually skipped)
 - [x] Dashboard integration
 - [x] Tests for non-logged in users
 - [x] Students who pass the course
 - [x] Students who don't pass the course

### Related PRs
- Deprecates: https://github.com/Edraak/edx-certificates/pull/1
- Theme: https://github.com/Edraak/edraak-platform-theme/pull/4

### CODE REVIEW

**Comment by OmarIthawi** [comment]
@Salomari1987 Could you please take a look?

**Comment by OmarIthawi** [comment]
Last ping @ahmedaljazzar and @Salomari1987. I'll merge by the end of day, if no one steps in.

File: lms/djangoapps/edraak_certificates/generator.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+            self.draw_bidi_text(self.course_org_disclaimer(), x, 1.44, size=0.16)
+        else:
+            self.draw_english_text(self.course_org_disclaimer(), x, 1.44, size=0.16)
+
+        self.draw_bidi_center_text(self.course_end_date, date_x, 4.82, size=0.27)
+
+        self.save()
+
+
+def generate_certificate(request, course):
```
Comment (iamjazzar) [inline lms/djangoapps/edraak_certificates/generator.py:296]: Do you think it's a good idea to send a request object to a non-view function?

Comment (OmarIthawi) [reply lms/djangoapps/edraak_certificates/generator.py:296]: I hate to do this, but when I need to call an edX function that requires it, I have no other solution.

**Comment by OmarIthawi** [comment]
@ahmedaljazzar any other comments?

File: lms/djangoapps/edraak_certificates/views.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+@login_required
+def issue(request, course_id):
+    return render_to_response('edraak_certificates/issue.html', {
+        'course_id': course_id,
+    })
+
+
+@transaction.non_atomic_requests
+@login_required
+def view(request, course_id):
```
Comment (iamjazzar) [inline lms/djangoapps/edraak_certificates/views.py]: Rename it please to something else.

File: lms/djangoapps/edraak_certificates/utils.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+    return the_decorator
+
+
+def is_certificates_feature_enabled():
+    if not settings.FEATURES.get('EDRAAK_CERTIFICATES_APP'):
+        return False
+
+    if not settings.FEATURES.get('ORGANIZATIONS_APP'):
+        raise Exception('Please enable the organization app (`ORGANIZATIONS_APP`) in '
```
Comment (iamjazzar) [inline lms/djangoapps/edraak_certificates/utils.py]: Can we change the exception message here to a more descriptive one like you enabled X without enabling Y.

File: requirements/edx/base.txt [hunk 4]
```diff
@@ -199,5 +199,9 @@ edx-django-release-util==0.2.0
 # modulestore inspection
 py2neo==3.1.2
 
+# Used for printing the Edraak certificates
+python-bidi==0.3.4
```
Comment (iamjazzar) [inline requirements/edx/base.txt:203]: You can use my [python-arabic-reshaper](https://github.com/ahmedaljazzar/python-arabic-reshaper) if you want.

**Review by iamjazzar — CHANGES_REQUESTED** [review]

File: lms/djangoapps/edraak_certificates/views.py (hunk 2 above)
Comment (OmarIthawi) [reply lms/djangoapps/edraak_certificates/views.py]: It's now `preview`.

File: requirements/edx/base.txt (hunk 4 above)
Comment (OmarIthawi) [reply requirements/edx/base.txt:203]: As much as I liked to use it, It has a bug:
```
ValueError: Invalid configuration: A section with the name ArabicReshaper was not found
```

I'm skipping this one.

File: lms/djangoapps/edraak_certificates/utils.py (hunk 3 above)
Comment (OmarIthawi) [reply lms/djangoapps/edraak_certificates/utils.py]: Done

File: lms/djangoapps/edraak_certificates/views.py (hunk 2 above)
Comment (OmarIthawi) [reply lms/djangoapps/edraak_certificates/views.py]: Done

**Comment by OmarIthawi** [comment]
@ahmedaljazzar I've resolved your comments. Please add your checkmark here, or `dismiss` your review.

**Review by iamjazzar — APPROVED** [review]

**Pull request merged**